File: app/app/src/main/java/com/cabal/app/Utils/AfterRegisterUserData.java
package com.cabal.app.Utils;

import androidx.annotation.NonNull;

import java.util.Arrays;

public class AfterRegisterUserData {
    private String avatar;
    private String nickname;
    private int searchRadius;
    private int[] interestIds;

    public int getSearchRadius() {
        return searchRadius;
    }

    public void setSearchRadius(int searchRadius) {
        this.searchRadius = searchRadius;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public int[] getInterestIds() {
        return interestIds;
    }

    public void setInterestIds(int[] interestIds) {
        this.interestIds = interestIds;
    }

    public AfterRegisterUserData(String avatar, String nickname, int[] interestIds, int searchRadius) {
        this.avatar = avatar;
        this.nickname = nickname;
        this.interestIds = interestIds;
        this.searchRadius = searchRadius;
    }

    @NonNull
    @Override
    public String toString() {
        return "AfterRegisterUserData{" +
                "avatar length='" + avatar.length() + '\'' +
                ", nickname='" + nickname + '\'' +
                ", interestIds=" + Arrays.toString(interestIds) +
                '}';
    }
}

File: app/app/src/main/java/com/cabal/app/utils/AfterRegisterUserData.kt
package com.cabal.app.utils

data class AfterRegisterUserData(val avatar: String, val nickname: String, val interestIds: IntArray, val searchRadius: Int) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AfterRegisterUserData

        if (avatar != other.avatar) return false
        if (nickname != other.nickname) return false
        if (searchRadius != other.searchRadius) return false
        if (!interestIds.contentEquals(other.interestIds)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = avatar.hashCode()
        result = 31 * result + nickname.hashCode()
        result = 31 * result + searchRadius
        result = 31 * result + interestIds.contentHashCode()
        return result
    }
}